### PR TITLE
Howto for rootless image building from OpenShift

### PR DESCRIPTION
**Pull request by jskovjyskebankdk**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Document how to get rootless image building working in the context of OpenShift.

#### How to verify it

Follow the tutorial and test that it actually works :)

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat WDYT?

File: docs/tutorials/05-openshift-rootless-bud.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
+![buildah logo](https://cdn.rawgit.com/containers/buildah/master/logos/buildah-logo_large.png)
+
+# Buildah Tutorial 5
+## Using Buildah to build images in a rootless OpenShift container
+
+This tutorial will walk you through setting up a container in OpenShift for building images.
+
+The instructions have been tested on OpenShift 4.3.27 with buildah 1.14.8.
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/05-openshift-rootless-bud.md:8]:
```suggestion
The instructions have been tested on OpenShift 4.3.27 with Buildah 1.14.8.
```

File: docs/tutorials/05-openshift-rootless-bud.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+          securityContext:
+            capabilities:
+              drop:
+                - KILL
+EOF
+
+deploymentconfig.apps.openshift.io/buildah created
+````
+
+#### The buildah container
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/05-openshift-rootless-bud.md:178]:
```suggestion
#### The Buildah container
```

File: docs/tutorials/05-openshift-rootless-bud.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+ secure-noroot: no (unlocked)
+ secure-no-suid-fixup: no (unlocked)
+ secure-keep-caps: no (unlocked)
+ secure-no-ambient-raise: no (unlocked)
+uid=1000(build)
+gid=1000(build)
+groups=
+````
+
+This is what the buildah data should look like:
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/05-openshift-rootless-bud.md:205]:
```suggestion
This is what the Buildah data should look like:
```

**Comment by TomSweeneyRedHat** [comment]
Apologies for not getting to this sooner.  I came back from PTO last week, saw this email, but lost it in the disaster my inbox was before reviewing.

A few nitty kinds of comments.  Buildah is "Buildah" in documentation unless the actual command is being invoked.  I noted the ones I found, if I missed any, please let me know.

I do not have an OC cluster handy, so I read the tutorial, but did not actually test it.  Have you tested it all by literally cutting/pasting (adjusting for un/pwd types of things) from this tutorial?

**Comment by rhatdan** [comment]
@jskovjyskebankdk are you still working on this, or would you like us to take it over?

**Comment by jskovjyskebankdk** [comment]
@rhatdan I will have a look later this week. Just back from vacation.

**Comment by jskovjyskebankdk** [comment]
@TomSweeneyRedHat I fixed the Buildah issues, and double checked that I can copy/paste my way through the instructions without problems.

**Comment by TomSweeneyRedHat** [comment]
Thanks a bunch @jskovjyskebankdk !  
LGTM
@rhdan or @nalind PTAL and merge if it LGTY

**Pull request merged**